pkg/image/oci/registry_provider.go

```diff
@@ -0,0 +1,92 @@
+package oci
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/anchore/stereoscope/internal/log"
+	"github.com/anchore/stereoscope/pkg/file"
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// RegistryImageProvider is a image.Provider capable of fetching and representing a container image fetched from a remote registry (described by the OCI distribution spec).
+type RegistryImageProvider struct {
+	imageStr        string
+	tmpDirGen       *file.TempDirGenerator
+	registryOptions *image.RegistryOptions
+}
+
+// NewProviderFromRegistry creates a new provider instance for a specific image that will later be cached to the given directory.
+func NewProviderFromRegistry(imgStr string, tmpDirGen *file.TempDirGenerator, registryOptions *image.RegistryOptions) *RegistryImageProvider {
```

Comment:
Why is this called `NewProviderFromRegistry`? I thought the convention was to name it like the struct it creates?

Reply:
good call, I'll adjust 👍